app/src/main/cpp/ExternalVR.cpp

```diff
@@ -172,8 +178,12 @@ struct ExternalVR::State {
 
   void PullBrowserStateWhileLocked() {
     const bool wasPresenting = IsPresenting();
-    memcpy(&browser, &data.browserState, sizeof(mozilla::gfx::VRBrowserState));
 
+    if (sourceBrowser == ExternalVR::VRBrowserType::Gecko) {
```

Comment:
The code may be better if we create something like `browserMutex` and `browserCond` references in `ExternalVR::State` and set them in `SetSourceBrowser`. That way we can avoid the ifs in all the Pull/push methods and just use the current ones from the state struct.